tests/Game.class.test.js

```diff
@@ -0,0 +1,916 @@
+'use strict';
+
+const Game = require('../src/modules/Game.class');
+const {
+  checkIsCorrectlyAligned,
+  getCellFromState,
+  transposeState,
+  cloneState,
+} = require('./test.helpers');
+
+describe('Game class', () => {
+  it('should be a function', () => {
+    expect(typeof Game).toBe('function');
+  });
+
+  it('should have a constructor', () => {
+    expect(typeof Game.prototype.constructor).toBe('function');
+  });
+
+  describe('core methods', () => {
+    let game2048;
+
+    beforeEach(() => {
+      game2048 = new Game();
+    });
+
+    [
+      'getState',
+      'getScore',
+      'getStatus',
+      'moveLeft',
+      'moveRight',
+      'moveUp',
+      'moveDown',
+      'start',
+      'restart',
+    ].forEach((method) => {
+      it(`should have a "${method}" method`, () => {
+        expect(typeof game2048[method]).toBe('function');
+      });
+    });
+  });
+
+  describe('initial state', () => {
+    it('should have a state of an empty board by default', () => {
+      const game2048 = new Game();
+      const state = game2048.getState();
+
+      expect(state).toEqual([
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+      ]);
+    });
+
+    it('should have a score of 0 by default', () => {
+      const game2048 = new Game();
+
+      expect(game2048.getScore()).toBe(0);
+    });
+
+    it('should have a status of "idle" by default', () => {
+      const game2048 = new Game();
+
+      expect(game2048.getStatus()).toBe('idle');
+    });
+
+    it('should have a state of a custom board if the initial state is provided', () => {
+      const initialState = [
+        [2, 0, 0, 0],
+        [0, 4, 0, 0],
+        [0, 0, 8, 0],
+        [0, 0, 0, 16],
+      ];
+
+      const game2048 = new Game(cloneState(initialState));
+      const state = game2048.getState();
+
+      expect(state).toEqual(initialState);
+    });
+  });
+
+  describe('after start', () => {
+    it('should have a score of 0', () => {
+      const game2048 = new Game();
+
+      game2048.start();
+
+      expect(game2048.getScore()).toBe(0);
+    });
+
+    it('should have a status of "playing"', () => {
+      const game2048 = new Game();
+
+      game2048.start();
+
+      expect(game2048.getStatus()).toBe('playing');
+    });
+
+    it('should have a state with two cells with values 2 or 4', () => {
+      const game2048 = new Game();
+
+      game2048.start();
+
+      const flatState = game2048.getState().flat();
+      const nonEmptyCells = flatState.filter((cell) => cell !== 0);
+
+      expect(nonEmptyCells).toHaveLength(2);
+
+      expect(
+        nonEmptyCells.every((cell) => [2, 4].includes(cell)),
+      ).toBe(true);
+    });
+
+    it('should generate the board with 4s less often than with 2s', () => {
+      const game2048 = new Game();
+
+      let statesCombined = [];
+      let twos = 0;
+      let fours = 0;
+
+      for (let i = 0; i < 100; i++) {
+        game2048.start();
```

Comment:
have we already tested that the second `start` call clears the prev state and creates a new one?